src/crimson/osd/ops_executer.cc

```diff
@@ -1183,9 +1184,16 @@ static PG::interruptible_future<ceph::bufferlist> do_pgnls_common(
   }
 
   return backend.list_objects(lower_bound, limit).then_interruptible(
-    [&backend, filter, nspace](auto&& ret)
+    [&backend, filter, nspace, &pg](auto&& ret)
     -> PG::interruptible_future<std::tuple<std::vector<hobject_t>, hobject_t>> {
       auto& [objects, next] = ret;
+      auto is_snapmapper = [&pg](const hobject_t &obj) {
+	if (obj == pg.make_snapmapper_oid().hobj) {
```

Comment:
Instead of adding pg ref parameter we can introduce `static bool PG::is_snapmapper_oid()`.
```
static inline bool is_snapmapper_oid(const hobject_t &obj) {
  return (obj.hobj.oid.name == "snapmapper");
}
```

Reply:
Done.

Reply:
Thanks, will schedule test after make check passes